src/System.Text.Json/tests/Utf8JsonReaderTests.cs

```diff
@@ -959,6 +963,9 @@ public static void Allow(string jsonString, string expectedComment, int expected
         [InlineData("\"h\u6F22\u5B57ello\"//This is a \u6F22\u5B57comment after json with new line\n", "//This is a \u6F22\u5B57comment after json with new line\n", 64)]
         [InlineData("{\"a\u6F22\u5B57ge\" : \n//This is a \u6F22\u5B57comment between key-value pairs\n 30}", "//This is a \u6F22\u5B57comment between key-value pairs\n", 66)]
         [InlineData("{\"a\u6F22\u5B57ge\" : 30//This is a \u6F22\u5B57comment between key-value pairs on the same line\n}", "//This is a \u6F22\u5B57comment between key-value pairs on the same line\n", 84)]
+        [InlineData("\"a\u6F22\u5B57lpha\" \r\n//This is a comment with a carriage return\r//Another single-line comment", "//This is a comment with a carriage return\r", 59)]
+        [InlineData("\"a\u6F22\u5B57lpha\" \r\n//This is a comment with a line break\n//Another single-line comment", "//This is a comment with a line break\n", 54)]
```

Comment:
Interesting test case where a single line comment doesn't end with any newline delimiter but rather end-of-payload (`//Another single-line comment`). We should check that we return the right thing when the user asks for this comment (when CommentHandling = Allow).

Reply:
Will add a test case based on this to #35705